docs/cartesian-product.adoc

```diff
@@ -300,37 +300,190 @@ To demonstrate with a table:
+To create the test data, instantiate a `new Sets()` then use the `add()` method to register the values for the parameters.
+For convenience, all `add()` methods return with your `Sets` instance, so you can chain `add()` calls.
+
+Let's look at an example.
+
+[source,java]
+----
+
+class MyCartesianTestClass {
+	@CartesianTest
+	@CartesianTest.Factory("setFactory")
+	void testMethod(String string, Class<?> clazz, TimeUnit unit) {
+		// passing test code
+	}
+
+	static Sets setFactory() {
+		return new Sets()
+			.add("Alpha", "Omega")
+			.add(Runnable.class, Cloneable.class, Predicate.class)
+			.add(TimeUnit.DAYS, TimeUnit.HOURS);
+	}
+}
+----
+
+The test `testMethod` is executed exactly twelve times.
+The first parameter can have any of the two values `"Alpha"` or `"Omega"`.
+The second parameter can have any of the three values `Runnable.class`, `Cloneable.class` or `Predicate.class`.
+The third parameter can have any of the two values `TimeUnit.DAYS` or `TimeUnit.HOURS`.
+`@CartesianTest` tests for all input combinations, that's `2 × 3 × 2`, so twelve tests in total.
+
+To demonstrate with a table:
+
+|===
+| # of test  | value of `string` | value of `clazz` | value of `unit`
+| 1st test   | "Alpha"           | Runnable.class   | TimeUnit.DAYS
+| 2nd test   | "Alpha"           | Runnable.class   | TimeUnit.HOURS
+| 3rd test   | "Alpha"           | Cloneable.class  | TimeUnit.DAYS
+| 4th test   | "Alpha"           | Cloneable.class  | TimeUnit.HOURS
+| 5th test   | "Alpha"           | Predicate.class  | TimeUnit.DAYS
+| 6th test   | "Alpha"           | Predicate.class  | TimeUnit.HOURS
+| 7th test   | "Omega"           | Runnable.class   | TimeUnit.DAYS
+| 8th test   | "Omega"           | Runnable.class   | TimeUnit.HOURS
+| 9th test   | "Omega"           | Cloneable.class  | TimeUnit.DAYS
+| 10th test  | "Omega"           | Cloneable.class  | TimeUnit.HOURS
+| 11th test  | "Omega"           | Predicate.class  | TimeUnit.DAYS
+| 12th test  | "Omega"           | Predicate.class  | TimeUnit.HOURS
+|===
+
+You can reuse the same argument provider method multiple times.
+
+[source,java]
+----
+class MyCartesianTestClass {
+	@CartesianTest
+	@CartesianTest.Factory("provideArguments")
+	void testNeedingArguments(String string, int i) {
+		// passing test code
+	}
+
+	@CartesianTest
+	@CartesianTest.Factory("provideArguments")
+	void testNeedingSameArguments(String string, int i) {
+		// different passing test code
+	}
+
+	static CartesianProductTest.Sets provideArguments() {
+		return new CartesianProductTest.Sets()
+			.add("Mercury", "Earth", "Venus")
+			.add(1, 12, 144);
+	}
+}
+----
+
+==== Conditions for the static factory method
+
+There are multiple conditions the static factory method has to fulfill to qualify:
+
+- must have the same name as the test method (or its name must be specified via the `factory` attribute)
+- must be `static`
+- must have **no** parameters
```

Comment:
What if we could inject parameters into factory methods, though?! 🤡

Reply:
Ha-ha, very funny.